strawberryfields/circuitspecs/x.py

```diff
@@ -0,0 +1,260 @@
+# Copyright 2019-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Circuit class specification for the X class of circuits."""
+
+import numpy as np
+from thewalrus.quantum import Amat
+from thewalrus.symplectic import expand
+
+import strawberryfields as sf
+from strawberryfields.decompositions import takagi
+from strawberryfields.program_utils import CircuitError, Command
+import strawberryfields.ops as ops
+
+from .circuit_specs import CircuitSpecs
+from .gbs import GBSSpecs
+from .gaussian_unitary import GaussianUnitary
+
+
+class Range:
```

Comment:
I think we should move `Range` and `Ranges` to the base `circuitspecs/circuit_specs.py` file, since they are quite general and might be used in other compilers.

Then, we can just do `from .circuit_specs import CircuitSpecs, Range, Ranges` here.